app/src/main/java/camp/nextstep/edu/github/di/MainComponent.kt

```diff
@@ -0,0 +1,24 @@
+package camp.nextstep.edu.github.di
+
+import camp.nextstep.edu.github.MainActivity
+import camp.nextstep.edu.github.data.di.RepositoryModule
+import camp.nextstep.edu.github.data.di.RetrofitModule
+import dagger.Subcomponent
+import javax.inject.Singleton
+
+/**
+ * android-github
+ * @author jaesung
+ * @created 2023/04/24
+ */
+@Singleton
```

Comment:
SubComponent를 만들었다는 뜻은 부모 Component와 생명주기를 다르게 하고싶다는 뜻으로 이해하고 있는데요! MainConponent의 루트는 AppComponent로 Application의 생명주기(@Singleton)으로 주로 사용된다고 생각해요. 그래서 `@Singleton` 대신 자체 스코프를 만드는 것은 어떠신가요?

예를 들면 `MainScope`가 간단하게 지을 수 있을 것 같네요?
```suggestion
@MainScope
```